### PR TITLE
Tjm 48 log in

### PR DESCRIPTION
**Pull request by Alvolutionor**

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -158,7 +160,15 @@ const createUser = async (req, res) => {
 
     return res.status(StatusCodes.OK).json(ret);
   } catch (err) {
-    return res.status(StatusCodes.BAD_REQUEST).json(err);
+    if (err.code === 11000) {
+      if (Object.keys(err.keyValue).includes('name')) {
+        return res.status(StatusCodes.CONFLICT).send('Name existed');
```
Comment (TomZhong-au) [inline controllers/user.js:165]: .json ({ message: 'Name existed' })

File: controllers/user.js [hunk 2]
```diff
@@ -158,7 +160,15 @@ const createUser = async (req, res) => {
...
+      }
+      if (Object.keys(err.keyValue).includes('email')) {
+        return res.status(StatusCodes.CONFLICT).send('Email registered');
```
Comment (TomZhong-au) [inline controllers/user.js:168]: .json ({ message: 'Email registered' })

File: controllers/user.js [hunk 3]
```diff
@@ -265,13 +275,32 @@ const toggleFollowing = async (req, res) => {
...
+    const user = await User.findOne({ email });
+    if (user.password === password) {
+      const data = { userId: user._id };
+      const token = jwt.sign(data, process.env.JWT_SECRET, {
+        algorithm: 'HS256',
+        expiresIn: process.env.JWT_EXPIRE_TIME,
+      });
+      res.cookie('token', token);
+      console.log(token);
+      return res.status(StatusCodes.OK).send('Successfully logged in');
```
Comment (TomZhong-au) [inline controllers/user.js:290]: .json ({ message: 'successfuly logged in' })

File: controllers/user.js [hunk 4]
```diff
@@ -265,13 +275,32 @@ const toggleFollowing = async (req, res) => {
...
+    }
+    return res.status(StatusCodes.UNAUTHORIZED).send('Wrong password, try again');
```
Comment (TomZhong-au) [inline controllers/user.js:292]: .json ({ message: 'wrong password' })

**Comment by Alvolutionor** [comment]
Fixed @TomZhong-au

**Review by Rachel-Fang — APPROVED** [review]

**Pull request merged**